docs/tutorials/quick-start/quick-start-part-3.md

```diff
@@ -0,0 +1,808 @@
+
+:::caution
+
+**Redux actions and state should only contain plain JS values like objects, arrays, and primitives. Don't put class instances, functions, or other non-serializable values into Redux!**.
+
+:::
+
+Since we can't just put a `Date` class instance into the Redux store, we'll track the `post.date` value as a timestamp string:
+
+```js title="features/posts/postsSlice.js"
+    postAdded: {
+      reducer(state, action) {
+        state.push(action.payload)
+      },
+      prepare(title, content, userId) {
+        return {
+          payload: {
+            id: nanoid(),
+            // highlight-next-line
+            date: new Date().toISOString(),
+            title,
+            content,
+            user: userId,
+          },
+        }
+      },
+    },
+```
+
+Like with post authors, we need to show the relative timestamp description in both our `<PostsList>` and `<SinglePostPage>` components. We'll add a `<TimeAgo>` component to handle formatting a timestamp string as a relative description. Libraries like `date-fns` have some useful utility functions for parsing and formatting dates, which we can use here:
+
+```jsx title="features/posts/TimeAgo.js"
+import React from 'react'
+import { parseISO, formatDistanceToNow } from 'date-fns'
+
+export const TimeAgo = ({ timestamp }) => {
+  let timeAgo = ''
+  if (timestamp) {
+    const date = parseISO(timestamp)
+    const timePeriod = formatDistanceToNow(date)
+    timeAgo = `${timePeriod} ago`
+  }
+
+  return (
+    <span title={timestamp}>
+      &nbsp; <i>{timeAgo}</i>
+    </span>
+  )
+}
+```
+
+### Sorting the Posts List
+
+Our `<PostsList>` is currently showing all the posts in the same order the posts are kept in the Redux store. Our example has the oldest post first, and any time we add a new post, it gets added to the end of the posts array. That means the newest post is always at the bottom of the page.
+
+Typically, social media feeds show the newest posts first, and you scroll down to see older posts. Even though the data is being kept oldest-first in the store, we can reorder the data in our `<PostsList>` component so that the newest post is first. In theory, since we know that the `state.posts` array is already sorted, we _could_ just reverse the list. But, it's better to go ahead and sort it ourselves just to be sure.
+
+Since `array.sort()` mutates the existing array, we need to make a copy of `state.posts` and sort that copy. We know that our `post.date` fields are being kept as date timestamp strings, and we can directly compare those to sort the posts in the right order:
+
+```jsx title="features/posts/PostsList.js"
+// Sort posts in reverse chronological order by datetime string
+//highlight-start
+const orderedPosts = posts.slice().sort((a, b) => b.date.localeCompare(a.date))
```

Comment:
Isn't it better to do this calculation in selectors and pass already prepared data to a component?
It seems like we will do all those calculation in run-time, so could be smth like this -
```ts
const selectExcerptPosts = state => {
  const posts = selectPosts(state);
  const orderedPosts = posts
    .slice()
    .sort((a, b) => b.date.localeCompare(a.date));

  return orderedPosts.map(orderedPost => {
    const date = parseISO(timestamp);
    const timePeriod = formatDistanceToNow(date);

    return {
      user: orderedPost.user,
      content: orderedPost.content.substring(0, 100),
      timePeriod,
      timeStamp: orderedPost.postDate
    };
  });
};
```
Later we could apply to reselect here to get the most benefit from selectors

Reply:
Yes and no.

We can't do it in a "plain" selector, as you've showed there, because we're creating a new array reference every time.  If we did this in a selector:

```js
const orderedPosts = useSelector( state => {
  const orderedPosts = state.posts.slice().sort();
  return orderedPosts;
})
```

that would force this component to re-render on _every_ dispatched action, which is not at all what we want.

We _could_ do that in a selector if we were using Reselect.  However, I haven't gotten to that part yet :)  That will come later.